src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,26 @@
+package calculator
+
+private typealias OperatorPair = Pair<IntArithmetics, Int>
+
+class Calculator(private val input: String) {
+    fun calculate(): Int {
+        val list = input.split(" ")
+        val identity = list[0].toInt()
+        return list.subList(1, list.size)
+            .chunked(2)
+            .map { chunk2Pair(it) }
+            .fold(identity) { acc, curr -> accumulate(acc, curr) }
```

Comment:
이런식으로 메소드로 한번 분리하고, :: 를 이용하여 메소드 레퍼런스형식으로 처리가 가능합니다!
```suggestion
            .fold(identity, ::accumulate)
```

```
fun accumulate(num: Int, set: Pair<IntArithmetics, Int>): Int {
        return set.first.apply(num, set.second)
}
```

Reply:
@vagabond95 
아항 저런 문법을 원했습니다. 감사합니다!
이런건 kotlin 공식 문서에서 뭐라고 검색해야 나오나요?

Reply:
검색 키워드: [Higher Order Function](https://play.kotlinlang.org/byExample/04_functional/01_Higher-Order%20Functions)